frontend/src/components/dialogs/AssetCompareDialog.vue

```diff
@@ -0,0 +1,158 @@
+<template>
+    <ff-dialog
+        ref="dialog" :header="header" confirm-label="Close" :closeOnConfirm="true" data-el="flow-view-dialog"
+        boxClass="!min-w-[80%] !min-h-[80%] !w-[80%] !h-[80%]" contentClass="overflow-hidden flex-grow"
+        @confirm="confirm()"
+    >
+        <template #default>
+            <div class="flex gap-2" data-el="snapshot-compare-toolbar">
+                <ff-dropdown v-model="compareSnapshot" class="flex-grow">
+                    <ff-dropdown-option
+                        v-for="option in compareSnapshotList" :key="option.value" :value="option.value"
+                        :label="option.label" :title="option.description" class="text-sm"
+                    >
+                        {{ option.label }}
+                    </ff-dropdown-option>
+                </ff-dropdown>
+                <ff-button
+                    v-if="true"
+                    :disabled="!compareSnapshot"
+                    data-action="compare-snapshots"
+                    kind="secondary"
+                    style="height: 30px; width: 106px"
+                    class="w-32"
+                    @click="renderComparison"
+                >
+                    Compare
+                </ff-button>
+            </div>
+            <div v-if="changes.length" class="flex justify-between items-center gap-2 mt-2 ml-2">
+                <div class="whitespace-nowrap">Change {{ changeIndex + 1 }} of {{ changes.length }}:</div>
+                <div class="text-sm text-gray-500 flex-grow truncate overflow-ellipsis">
+                    {{
+                        changes[changeIndex].toString()
+                    }}
+                </div>
+                <ff-button kind="secondary" size="small" class="w-14" @click="gotoPreviousDifference">Prev</ff-button>
+                <ff-button kind="secondary" size="small" class="w-14" @click="gotoNextDifference">Next</ff-button>
+            </div>
+            <div v-else class="mt-2">
+                <div class="text-sm text-gray-500 flex-grow truncate overflow-ellipsis ml-2">No differences found</div>
+            </div>
+            <div ref="compareViewer" data-el="ff-flow-compare-view" class="ff-flow-compare-viewer pt-4" @click.stop.prevent>
+              &nbsp;
+            </div>
+        </template>
+        <template #actions>
+            <div class="flex justify-end">
+                <ff-button data-action="dialog-confirm" @click="confirm()">Close</ff-button>
+            </div>
+        </template>
+    </ff-dialog>
+</template>
+<script>
+
+import FlowRenderer from '@flowfuse/flow-renderer'
+
+import SnapshotsApi from '../../api/snapshots.js'
+
+import Alerts from '../../services/alerts.js'
+
+export default {
+    name: 'FlowViewerDialog',
```

Comment:
this is same name as the other dialog (causes confusion when debugging)

Reply:
It must have slipped by me when I duplicated the component. I'll fix it